component/common/loki/client/batch.go

```diff
@@ -48,6 +52,16 @@ func newBatch(maxStreams int, entries ...loki.Entry) *batch {
 	return b
 }
 
+type SendDataMarkerHandler interface {
```

Comment:
Yes for sure, will work through them. Didn't add much docs on top of those since this PR is branched of the other one. Now that https://github.com/grafana/agent/pull/5434 is approved I'll rebase and clean up this one (docs, tests, benches, etc).

Reply:
Tests, comments and bench added